src/main/scala/io/iohk/ethereum/merklepatriciatrie/Node.scala

```diff
@@ -0,0 +1,102 @@
+package io.iohk.ethereum.merklepatriciatrie
+
+import io.iohk.ethereum.rlp.{decode => decodeRLP, encode => encodeRLP, _}
+import io.iohk.ethereum.merklepatriciatrie.MerklePatriciaTrie.HashFn
+
+/**
+  * Trie elements
+  */
+private[merklepatriciatrie] sealed trait Node {
+
+  import MerklePatriciaTrie._
+
+  private var hashedCache: Option[Array[Byte]] = None
```

Comment:
@adamsmo when you are working with a node you might need to access several times to it's hash. Either to store the reference inside another node or to use it as a key to store it in the DataSource.